tests/PullRequest.py

```diff
@@ -404,3 +409,50 @@ def testUpdateBranch(self):
             self.pull.update_branch("addaebea821105cf6600441f05ff2b413ab21a36")
         )
         self.assertTrue(self.pull.update_branch())
+
+    def testDeleteOnMerge(self):
+        self.assertTrue(
+            self.delete_restore_repo.get_branch(self.delete_restore_pull.head.ref)
+        )
+        self.assertFalse(self.delete_restore_pull.is_merged())
+        status = self.delete_restore_pull.merge(deletebranch=True)
+        self.assertTrue(status.merged)
+        self.assertTrue(self.delete_restore_pull.is_merged())
+        self.assertRaises(
```

Comment:
Can you use with assertRaises() so we can see the content of the exception? There are a few other examples in the code base.